lib/src/core/engine.dart

```diff
@@ -145,14 +148,14 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
       await events.waitFor<EnginePeerStateUpdatedEvent>(
         filter: (event) => event.isPrimary && event.state.isConnected(),
         duration: this.connectOptions.timeouts.connection,
-        onTimeout: () => throw ConnectException(),
+        onTimeout: () => throw ConnectException(
+            'Timed out waiting for EnginePeerStateUpdatedEvent'),
       );
 
       _updateConnectionState(ConnectionState.connected);
     } catch (error) {
       logger.fine('Connect Error $error');
       _updateConnectionState(ConnectionState.disconnected);
-      rethrow;
```

Comment:
does this mean errors are no longer reported to the caller? how does Room know that the connection had errored?

Reply:
https://github.com/livekit/client-sdk-flutter/pull/194/commits/17396100604570b50a833c0f54fe333a1b040dbd